### ISSUE TITLE
Add the ability for the binder to detect ChainedKafkaTransactionManager bean

### ISSUE THREAD

**Issue by sobychacko** [spring-cloud/spring-cloud-stream-binder-kafka#861]
Currently, the binder creates a default transaction manager internally. If an application creates a bean of type `ChainedKafkaTransactionManager`, then the binder should use that in order to properly synchronize with other transaction managers such as a JPA based one.

**Comment by garyrussell** [comment]
@sobychacko It turns out we don't need this; you can use a listener container customizer to inject a `ChainedTransactionManager` bean.

```java
@Bean
public ListenerContainerCustomizer<AbstractMessageListenerContainer<?, ?>> customizer() {
	return (container, dest, group) -> container.getContainerProperties().setTransactionManager(myChainedTm());
}
````

**Comment by garyrussell** [comment]
Actually, no; forget that.

**Comment by santhoshpt** [comment]
Is this fix available in spring-cloud-stream-binder-kafka:3.0.3.RELEASE?  I'm currently using this release and facing issue in writing chained transaction on JpaTM and KafkaTM.  The transaction used to work fine with spring boot 2.1.11 and dependent cloud libraries.  After upgrading the kafka consumer is able to receive the messages even if the jpa transaction is till under process.

**Comment by garyrussell** [comment]
Your question is not clear; there is no way to use a chained transaction manager with consumers before this fix is available.

I presume your question is related to https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/869.
